src/rcclReduceScatter.cpp

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright (c) 2017 - Present Advanced Micro Devices, Inc.
+All rights reserved.
+*/
+
+/**
+ * @file rcclReduceScatter.cpp
+ * @brief rccl library implementation of rcclReduceScatter API
+ *
+ * This file contains implementation of rcclReduceScatter API.
+ *
+ */
+
+#include "rcclDataTypes.h"
+#include "rcclHelper.h"
+#include "rcclSetKernels.h"
+#include "rcclTracker.h"
+
+#include "rcclScalarReduceScatterRuntime.h"
+
+#include <string>
+#include <unordered_map>
+
+extern std::unordered_map<int, std::string> umap_red_op;
+extern std::unordered_map<int, std::string> umap_datatype;
+
+extern int RCCL_TRACE_RT;
+
+//! @brief Definition of rcclReduceScatter
+rcclResult_t rcclReduceScatter(const void *sendbuff, void *recvbuff, int recvcount,
+                           rcclDataType_t datatype, rcclRedOp_t op,
+                           rcclComm_t comm, hipStream_t stream) {
+    if ((RCCL_TRACE_RT & krccl_print_api) == krccl_print_api) {
+        int dev;
+        hipGetDevice(&dev);
+        fprintf(stderr,
+                "%s<<rccl-api:%s rccl-device:%d sendbuff:%p recvbuff:%p "
+                "recvcount:%d datatype:%s op:%s comm:%p stream:%p%s\n",
+                API_COLOR, __func__, dev, sendbuff, recvbuff, recvcount,
+                umap_datatype[datatype].c_str(), umap_red_op[op].c_str(), comm,
+                stream, API_COLOR_END);
+    }
+
+    //! Check if buffer pointers are not null
+    if (sendbuff == nullptr || recvbuff == nullptr) {
+        return rcclInvalidDevicePointer;
+    }
+
+    //! Check if data type of buffers is valid or not
+    if (datatype >= rccl_NUM_TYPES) {
+        return rcclInvalidType;
+    }
+
+    //! Check if op is valid or not
+    if (op >= rccl_NUM_OPS) {
```

Comment:
Same issue as above.

Reply:
It might be worth filing a separate issue to fix all these checks.  I'm assuming this code was copy-and-pasted and modified based on the other implemented functions so these checks would appear elsewhere, as well.

Reply:
agreed. let's fix this globally in another patch.